packages/lib/emailSchema.ts

```diff
@@ -0,0 +1,6 @@
+import { z } from "zod";
+
+export const emailRegex =
```

Comment:
What is the source of this regex? Let's add it in the code.

Also, if that source doesn't have tests, we need the unit tests in here.

Reply:
The regex is sourced from https://github.com/colinhacks/zod/issues/3155#issuecomment-2060045794